src/Rifm.js

```diff
@@ -137,7 +147,24 @@ export const Rifm = (props: Props) => {
         // if nothing changed for formatted value, just refresh so userValue will be used at render
         refresh();
       } else {
-        props.onChange(formattedValue);
+        if (process.env.NODE_ENV !== 'production') {
+          const replaceValue =
+            props.replace != null
+              ? props.replace(formattedValue)
+              : formattedValue;
+
+          if (replaceValue.length !== formattedValue.length) {
+            console.warn('replace operation to work, must preserve length');
+          }
+
+          props.onChange(replaceValue);
+        } else {
+          props.onChange(
+            props.replace != null
+              ? props.replace(formattedValue)
+              : formattedValue
```

Comment:
Put replaceValue into upper scope and reuse it here please. It will be inlined by minifier.

Reply:
No, 10 bytes gzipped lost

Reply:
Readability of this fragment is bad. Only warning condition should be wrapped with NODE_ENV